### PR TITLE
Added linting

### PR DESCRIPTION
**Pull request by calvintam236**

### CODE REVIEW

**Comment by WieFel** [comment]
LGTM. But there are issues when merging into `master`, because `feature/lint` is not up-to-date.
Let's first merge `master` into `feature/lint` and resolve the issues + maybe already make the lint fixes.

**Comment by WieFel** [comment]
@calvintam236 Fixed all lint warnings. If you don't understand why I did something at some points, simply checkout commit 853df50 and run lint yourself, so you can see why I had to change something ;)

If you want, we can merge this first and then release v 2.0.0!?

File: lib/src/conversion.dart [hunk 1]
```diff
@@ -71,13 +71,16 @@ HSVColor rgbToHsv({
...
+}) =>
+    HSVColor.fromColor(Color.fromARGB(a, r, g, b));
 
 double _hue2rgb(double p, double q, double t) {
-  if (t < 0) t += 1;
-  if (t > 1) t -= 1;
-  if (t < 1 / 6) return p + (q - p) * 6 * t;
-  if (t < 1 / 2) return q;
-  if (t < 2 / 3) return p + (q - p) * (2 / 3 - t) * 6;
+  double tValue = t;
```
Comment (calvintam236) [inline lib/src/conversion.dart:78]: I think this t is tint

File: example/lib/main.dart [hunk 2]
```diff
@@ -32,8 +34,10 @@ class ExamplePage extends StatelessWidget {
         children: <Widget>[
           _createListItem(
             title: "Original color",
-            subtitle: """TinyColor tc = TinyColor(Colors.blue)
-Color c = Colors.blue""",
+            subtitle: """
+TinyColor tc = TinyColor(Colors.blue)
+Color c = Colors.blue
+            """,
```
Comment (calvintam236) [inline example/lib/main.dart:40]: this `”””` ending should have no spaces before it (start of line)

File: lib/src/tinycolor.dart [hunk 3]
```diff
@@ -160,6 +172,6 @@ class TinyColor {
   bool equals(Object other) => this == other;
 }
 
-extension _ on Color {
+extension ColorExtension on Color {
```
Comment (calvintam236) [inline lib/src/tinycolor.dart:175]: I think the name should start with `_`. It is not for public?

File: example/lib/main.dart (hunk 2 above)
Comment (WieFel) [reply example/lib/main.dart:40]: True. Could you quickly fix that please?

File: lib/src/tinycolor.dart (hunk 3 above)
Comment (WieFel) [reply lib/src/tinycolor.dart:175]: True. Please add `_`

**Review by WieFel — APPROVED** [review]

**Pull request merged**